components/pd_client/src/client.rs

```diff
@@ -526,6 +526,42 @@ impl PdClient for RpcClient {
             .boxed()
     }
 
+    fn get_buckets_by_id(&self, region_id: u64) -> PdFuture<Option<metapb::Buckets>> {
+        let timer = Instant::now();
+
+        let mut req = pdpb::GetRegionByIdRequest::default();
```

Comment:
Duplicated with `get_region_by_id`.

Reply:
`get_region_by_id ` returns ` PdFuture<Option<metapb::Region>>`.

Reply:
Yes, but the implementation is duplicated.

Reply:
fixed， extract the duplicated code as function.